kernelci/lab/shell.py

```diff
@@ -0,0 +1,33 @@
+import os
+import subprocess
+from jinja2 import Environment, FileSystemLoader
+from kernelci.lab import LabAPI
+
+
+class Shell(LabAPI):
+
+    def generate(self, params, device_config, plan_config,
+                 callback_opts=None, templates_path='config/scripts',
+                 db_config=None):
+        jinja2_env = Environment(loader=FileSystemLoader(
+            templates_path or self.TEMPLATES_PATH
+        ))
+        template_path = plan_config.get_template_path(None)
+        template = jinja2_env.get_template(template_path)
+        if db_config:
+            params['db_config_yaml'] = db_config.to_yaml()
+        return template.render(params)
+
+    def save_file(self, *args, **kwargs):
+        output_file = super().save_file(*args, **kwargs)
+        os.chmod(output_file, 0o775)
+        return output_file
+
+    def submit(self, job_path, get_process=False):
+        process = subprocess.Popen(job_path, shell=True)
```

Comment:
Is it needed here to set `shell=True`.  If it's not necessary to run the command in a shell we should consider setting it to False.

Reply:
Yes that's right, it's not required.  I've removed it now.